packages/expo-file-system/android/src/main/java/expo/modules/filesystem/next/FileSystemFileAdapter.kt

```diff
@@ -0,0 +1,101 @@
+package expo.modules.filesystem.next
+
+import android.content.Context
+import android.net.Uri
+import androidx.core.net.toUri
+import androidx.documentfile.provider.DocumentFile
+import java.io.File
+import java.net.URI
+
+class FileSystemFileAdapter(val context: Context, val uri: Uri) {
+  val javaFile get() = File(URI.create(uri.toString()))
+  val treeDocumentFile: DocumentFile? by lazy {
+    DocumentFile.fromTreeUri(context, uri)
+  }
+  val singleDocumentFile: DocumentFile? by lazy {
+    DocumentFile.fromSingleUri(context, uri)
+  }
+
+  // TODO: consider adding the SAF domain check here
+  val isContentURI = uri.scheme == "content"
+
+  fun exists(): Boolean {
+    return if (isContentURI) {
+      singleDocumentFile?.exists() == true
+    } else {
+      javaFile.exists() == true
```

Comment:
```suggestion
      javaFile.exists()
```
Checking if the bool value is true seems weird